src/build/build_step.go

```diff
@@ -1131,6 +1136,65 @@ func fetchOneRemoteFile(state *core.BuildState, target *core.BuildTarget, url st
 	return nil
 }
 
+// setHeaders sets up all the headers we should send on remote_file() requests, including User-Agent and any user
+// defined ones.
+func setHeaders(req *http.Request, target *core.BuildTarget, env core.BuildEnv) error {
+	req.Header.Set("User-Agent", fmt.Sprintf("please.build/%s", core.PleaseVersion))
+
+	param := func(str string) (string, string) {
+		if !strings.HasPrefix(str, "remote_file:") {
+			return "", ""
+		}
+		str = strings.TrimPrefix(str, "remote_file:")
+		i := strings.IndexRune(str, ':')
+		return str[:i], str[(i + 1):]
+	}
+
+	userName := ""
+	password := ""
+	for _, l := range target.Labels {
+		param, value := param(l)
+		switch param {
+		case "":
+			continue
```

Comment:
Does Go not allow an empty `case`?

Reply:
Yeah, that's what the default case below is. This is matching the case where `param()` returns an empty string, which it will do when the label isn't a parameter. The default case handles when we do something like :`remote_file:nonsense_param:blah` which is different.